### DISCUSSION TITLE
vega-view

### DISCUSSION THREAD

**Discussion by arnolddevos** [vega/vega-lite#9409] [Show and tell]
Hi, here is a simple visualisation tool called [vega-view](https://github.com/cuprous-au/vega-view) that I created. It displays [nushell](https://www.nushell.sh/) tables using [Vega Lite](https://vega.github.io/vega-lite/) in a webview.  vega-view operates from the command line and pops open a window for the visualisation.  It does not rely on the network or web browser.  

For example, if you have a source of data such as a file `example.json` the command `open example.json | vega view (vega bar b --category=t --subcategory=a)` pops up a window with a bar graph.  

The advantage of combining nushell with vega is that nushell can source and manipulate many forms of data which vega-lite can visualise in many different ways. 

But vega-view can be used without nushell to.  The core vega-view executable expects JSON data and a vega-lite specification to produce a visualisation in a window.

**Comment by domoritz** [comment]
Very cool. Want to send us a pull request to add this to https://vega.github.io/vega-lite/ecosystem.html?

**Reply by arnolddevos** [reply]
Done.  BTW I did not immediately find the ecosystem page.  I stumbled on https://github.com/joaopalmeiro/awesome-vega which does not seem to be in use.  Anyway, the ecosystem page is full of useful things.
